caffe2/sgd/adagrad_op.h

```diff
@@ -81,6 +62,26 @@ class SparseAdagradOp final : public Operator<Context> {
         epsilon_(OperatorBase::GetSingleArgument<float>("epsilon", 1e-5f)) {}
 
   bool RunOnDevice() override {
+    // Enforce types
+    CAFFE_ENFORCE(OperatorBase::InputIsType<Tensor<Context>>(PARAM));
+    CAFFE_ENFORCE(OperatorBase::InputIsType<Tensor<Context>>(MOMENT_1));
+    CAFFE_ENFORCE(OperatorBase::InputIsType<Tensor<Context>>(INDICES));
+    CAFFE_ENFORCE(OperatorBase::InputIsType<Tensor<Context>>(GRAD));
+    CAFFE_ENFORCE(OperatorBase::InputIsType<Tensor<Context>>(LR));
```

Comment:
no need to do those explicitly, error message would be generated anyway (and there was another PR recently that adds the input name to error message)